sklearn/preprocessing/_polynomial.py

```diff
@@ -54,14 +54,17 @@ class SplineTransformer(TransformerMixin, BaseEstimator):
           `degree` number of knots are added before the first knot, the same
           after the last knot.
 
-    extrapolation : {'error', 'constant', 'linear', 'continue'}, \
+    extrapolation : {'error', 'constant', 'linear', 'continue', 'periodic'}, \
         default='constant'
         If 'error', values outside the min and max values of the training
         features raises a `ValueError`. If 'constant', the value of the
         splines at minimum and maximum value of the features is used as
         constant extrapolation. If 'linear', a linear extrapolation is used.
         If 'continue', the splines are extrapolated as is, i.e. option
-        `extrapolate=True` in :class:`scipy.interpolate.BSpline`.
+        `extrapolate=True` in :class:`scipy.interpolate.BSpline`. If
+        'periodic', periodic splines with a periodicity equal to the distance
+        between the first and last knot are used. Periodic splines enforce
+        equal function values and derivatives at the first and last knot.
```

Comment:
I think we could be extra helpful and explain why it's a nice property on a practical use case. For instance:

```suggestion
        equal function values and derivatives at the first and last knot.
        For example, this makes it possible to avoid introducing an arbitrary
        jump between Dec 31st and Jan 1st in splines features derived from a
        naturally periodic "day-of-year" input feature. In this case it is
        recommend to manually set the knot values to control the period.
```

Maybe it's a bit verbose for a docstring but it took quite a long time for me to get it, I think it's better to be extra explicit and many of our users might find it useful.

Alternatively we could have a similar paragraph in the user guide (under `doc/modules/preprocessing.rst`).

Reply:
I don't mind verbosity at all.